dev/TEAL.md

```diff
@@ -0,0 +1,219 @@
+---
+numbersections: true
+title: "Algorand Transaction Execution Approval Language"
+date: \today
+abstract: >
+  Algorand allows transactions to be effectively signed by a small program. If the program evaluates to true then the transaction is allowed. This document defines the language and bytcode format.
+---
+
+# Transaction Execution Approval Language (TEAL)
+
+TEAL is a bytecode based stack language that executes inside Algorand transactions to check the parameters of the transaction and approve the transaction as if by a signature.
+
+TEAL programs should be short, at most 1000 bytes including all constants and operations, and run fast as they are run in-line along with signature checking, transaction balance rule checking, and other checks during block assembly and validation.
```

Comment:
Might want to parametrize "1000" here (e.g., `MaxProgramSize`).

Reply:
To make it uniform across the document, it might make more sense just to specify LogicSigMaxSize and LogicSigMaxCost somewhere in the beginning, and then just refer to those in the document. I guess if those are ever changed, then you can update fewer places in the document.

Reply:
I simplified this by making the opening paragraph more vague and introductory. Later details mention numbers and consensus parameters.